contracts/MechMarketplace.sol

```diff
@@ -371,27 +347,30 @@ contract MechMarketplace is IErrorsMarketplace {
         emit SetMechFactoryStatuses(mechFactories, statuses);
     }
 
-    // Check and escrow delivery rate
-    function _checkAndEscrowDeliveryRate(address mech) internal {
-        IMech.MechType mechType = IMech(mech).mechType();
-        uint256 maxDeliveryRate = IMech(mech).maxDeliveryRate();
+    /// @dev Sets mech type escrows.
+    /// @param mechTypes Mech types.
+    /// @param escrows Corresponding escrow addresses.
+    function setMechTypeEscrows(IMech.MechType[] memory mechTypes, address[] memory escrows) external {
```

Comment:
Each mech type has its own escrow contract to manage type-specific payments / fees.